mne/viz/backends/tests/test_renderer.py

```diff
@@ -152,3 +153,11 @@ def test_3d_backend(renderer):
                     focalpoint=center)
     rend.reset_camera()
     rend.show()
+
+
+def test_get_3d_backend(backend_mocker):
+    """Test get_3d_backend private function call for side-effects."""
+    # Test twice to ensure the first call had no side-effect
+    mne.viz.backends.renderer.MNE_3D_BACKEND = None
+    assert mne.viz._get_3d_backend() is None
+    assert mne.viz._get_3d_backend() is None
```

Comment:
I suggest to migrate to local import of `renderer` in the test to have better control over the backends. The function `get_3d_backend` will try all the valid backends in the list `VALID_3D_BACKENDS`, the default being `mayavi` so testing with `None` will not work. This is my suggestion:

```py
def test_get_3d_backend():
    """Test get_3d_backend private function call for side-effects."""
    # Test twice to ensure the first call had no side-effect
    from mne.viz.backends import renderer
    backend_name = renderer.MNE_3D_BACKEND
    assert renderer.get_3d_backend() == backend_name
    assert renderer.get_3d_backend() == backend_name
```

Reply:
Or using the `renderer` fixture:

```py
def test_get_3d_backend(renderer):
    """Test get_3d_backend private function call for side-effects."""
    # Test twice to ensure the first call had no side-effect
    backend_name = renderer.MNE_3D_BACKEND
    assert renderer.get_3d_backend() == backend_name
    assert renderer.get_3d_backend() == backend_name
```

Reply:
You are right. I only thought about this from the perspective of my test environment, where I had uninstalled mayavi and pyvista. Of course if they're available, my proposed test setup doesn't make too much sense. Will look into this this afternoon